### PR TITLE
Fix/falsy id 3.x

### PR DESCRIPTION
**Pull request by STRML**
### Description

Port of #3693 and #3719 into 3.x

Ping @bajtos - in particular, unsure how to get a 0 id set on User.create, appears new validation was added in 3.x.

Additionally it appears something is failing all PRs in your CI and we in the public lack visibility.

#### Related issues

#2356, #2374, #3130, #3693

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by dhmlau** [comment]
@slnode test please

**Comment by bajtos** [comment]
> Ping @bajtos - in particular, unsure how to get a 0 id set on User.create, appears new validation was added in 3.x.

In 3.0, we enabled `forceId` by default. This flag prevents clients from sending their own id, and requires the database to generate one.

You can take a look e.g. [here](https://github.com/strongloop/loopback/blob/1babfcde9fcece67b7e43851d767de0a04b1e802/test/user.test.js#L68-L70) to see how to create a User model with `forceId` disabled.

> Additionally it appears something is failing all PRs in your CI and we in the public lack visibility.

Yeah, the lack of visibility into cis-jenkins builds is a long standing pain point :(

The test "should be properly authenticated with 0 userId" failed with the following message:

```
The `User` instance is not valid. Details: `id` can't be set (value: 0).
ValidationError: The `User` instance is not valid. Details: `id` can't be set (value: 0).
  at node_modules/loopback-datasource-juggler/lib/dao.js:398:12
  at ModelConstructor.<anonymous> (node_modules/loopback-datasource-juggler/lib/validations.js:578:13)
  at ModelConstructor.next (node_modules/loopback-datasource-juggler/lib/hooks.js:93:12)
  at done (node_modules/loopback-datasource-juggler/lib/validations.js:575:25)
  at node_modules/loopback-datasource-juggler/lib/validations.js:653:7
  at node_modules/async-listener/glue.js:188:31
  at _combinedTickCallback (internal/process/next_tick.js:73:7)
  at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:128:9)
```

(I think you should see the same problem in Travis CI output.)

**Comment by STRML** [comment]
Fixed the test in the latest commit @bajtos.

File: test/role.test.js [hunk 1]
```diff
@@ -364,6 +364,64 @@ describe('role model', function() {
     });
   });
 
+  it.only('should be properly authenticated with 0 userId', function(done) {
```
Comment (bajtos) [inline test/role.test.js]: forgotten `it.only`

**Comment by STRML** [comment]
Yikes! Thanks.

**Comment by STRML** [comment]
Hold up, the it.only() is still in there.

**Comment by bajtos** [comment]
Oops, too late 😨

I remember seeing a commit removing `it.only`, was it in a different patch? Anyways, I guess we need to open a new pull request now - do you have bandwidth to do so yourself?

**Comment by STRML** [comment]
Yep can do and can add a lint rule too.

**Comment by bajtos** [comment]
> can add a lint rule too.

That would be awesome! Is there a way how to put that rule in our shared eslint config in https://github.com/strongloop/eslint-config-loopback?

**Pull request merged**